be/src/exec/pipeline/scan/chunk_source.cpp

```diff
@@ -65,14 +67,23 @@ Status ChunkSource::buffer_next_batch_chunks_blocking(RuntimeState* state, size_
 
             ChunkPtr chunk;
             _status = _read_chunk(state, &chunk);
+            // we always output a empty chunk instead of nullptr, because we need set tablet_id and is_last_chunk flag
+            // in the chunk.
```

Comment:
I don't think always output a empty chunk is reasonable.

Reply:
In some cases, tablet will output nullptr,  Query cache should take such a assumption that where is no nullptr ouput, for an example, Empty tablet output nullptr chunk.